lua/entities/gmod_tardis/modules/sh_destination.lua

```diff
@@ -92,22 +100,83 @@ TARDIS:AddKeyBind("destination-demat",{
     clientonly=true,
     exterior=true
 })
+TARDIS:AddKeyBind("destination-jump",{
+    name="Jump",
+    section="Destination",
+    func=function(self,down,ply)
+        if ply:GetTardisData("destination") then
+            if down then
+                self:SetData("destination-trace",true)
+            elseif self:GetData("destination-trace") then
+                local prop = self:GetData("destinationprop")
+                local pos,ang = self:GetDestinationPropTrace(ply,ply:EyeAngles())
+
+                prop:SetPos(pos)
+                prop:SetAngles(ang)
+
+                self:SetData("destination-trace",false)
+            end
+        end
+    end,
+    key=MOUSE_RIGHT,
+    clientonly=true,
+    exterior=true
+})
 TARDIS:AddKeyBind("destination-snaptofloor",{
     name="SnapToFloor",
     section="Destination",
     func=function(self,down,ply)
-        if ply:GetTardisData("destination") then
+        if down and ply:GetTardisData("destination") then
             local prop = self:GetData("destinationprop")
             if IsValid(prop) then
-                local pos = util.QuickTrace(prop:GetPos()+Vector(0,0,50), Vector(0,0,-1)*99999999).HitPos
+                local prevpos = self:GetData("destination_snaptofloor_lastpos")
+                local should_glue = (prevpos == prop:GetPos())
+
+                local pos, ang = self:GetGroundedPos(prop:GetPos(), should_glue)
```

Comment:
This position does not ignore the TARDIS itself like the right click does so you can snap on top of itself 
![image](https://github.com/MattJeanes/TARDIS/assets/2363642/d5b48f07-7e47-40f3-b9e9-6bc7af3ba7df)

Reply:
aee08d3b9afe0165cf4b2aefa4426c9a835fa4b7
will ignore itself, but not other TARDISes